gateway/utils/letsencrypt/sc_store.go

```diff
@@ -0,0 +1,122 @@
+package letsencrypt
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/mholt/certmagic"
+	"github.com/sirupsen/logrus"
+	apigo "github.com/spaceuptech/space-api-go"
+	"github.com/spaceuptech/space-api-go/db"
+	"github.com/spaceuptech/space-api-go/types"
+)
+
+type storage struct {
+	sync.RWMutex
+	db         *db.DB
+	collection string
+}
+
+func NewScStore(project, url, db string) *storage {
+	return &storage{db: apigo.New(project, url, false).DB(db), collection: "certificates"}
+}
+
+func (s *storage) Store(key string, value []byte) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	m := types.M{
+		"_id":   key,
+		"value": base64.StdEncoding.EncodeToString(value),
+		"size":  len(value),
+	}
+
+	data, err := s.db.Upsert(s.collection).Set(m).Apply(ctx)
+	if err != nil || data.Status != http.StatusOK {
+		logrus.Errorf("error while storing data of lets encrypt - %v", err)
+		return err
+	}
+	return nil
+}
+
+func (s *storage) Load(key string) ([]byte, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	data, err := s.db.GetOne(s.collection).Where(types.M{"_id": key}).Apply(ctx)
+	if err != nil || data.Status != http.StatusOK {
+		logrus.Errorf("error while retrieving data of lets encrypt - %v", err)
+		return nil, err
+	}
+	return json.Marshal(data.Data)
```

Comment:
You only need to return the value here. Not the entire object.

Reply:
Found a problem in `space-api-go`. Data will actually have a field named result which could contain an array or map depending on the operation